src/controller/directoryController.ts

```diff
@@ -100,6 +101,20 @@ export class DirectoryController {
     }
   }
 
+  /**
+   * Deletes the associated uri of all selected AddonTreeItems.
+   * @param list Selected AddonTreeItems
+   * @returns whether all were successfully deleted.
+   */
+  static async deleteUri(list: AddonTreeItem[]) {
+    let success = false;
+    list.forEach(async (item) => {
+      await vscode.workspace.fs.delete(item.uri, { recursive: true });
+      success = true;
+    });
+    return success;
```

Comment:
I don't think this is doing what is documented in the code comment above. `deleteUri` will throw if deletion of an `AddonTreeItem` is not possible. Otherwise, `success` will be set to `true` but this is equivalent to returning `true` unconditionally after the `forEach()`. There is no need for a variable. In fact, the behavior is not even consistent because deleting an empty list should be successful, and in this case, it would return `false` (because we won't enter the `forEach()` block).

Now to fix this problem, it depends on what you want to do. You could use [`Promise.allSettled()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) and determine whether all the tree items have been deleted by counting the number of "fulfilled" results. Or we don't return anything?